internal/controllers/machineset/msutil/names.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package msutil implements MachineDeployment utilities.
+package msutil
+
+import (
+	"encoding/base64"
+	"hash/fnv"
+)
+
+// MachineSetNameLabelValue returns the machineSetName if it meets the standards for a Kubernetes label value.
+// If the name can not be used as a label value this function returns a hash.
+func MachineSetNameLabelValue(machineSetName string) string {
+	if len(machineSetName) <= 63 {
+		return machineSetName
+	}
+	hasher := fnv.New32a()
+	var _, _ = hasher.Write([]byte(machineSetName)) // Can never return error.
```

Comment:
```suggestion
	_, _ := hasher.Write([]byte(machineSetName)) // Can never return error.
```

Or just remove the `_` and disable linter? We could even catch the error and panic instead, in that case change the function name to have the `Must` prefix

Reply:
Updated to use  `nolint:gosec`. Not sure what's the safest way to deal with functions like this which should never return an error despite having it in the signature.